livemaker/cli/lmlsb.py

```diff
@@ -1159,3 +1159,154 @@ def scan_menus(lsb, csv_data, file_name, extract=False, patch=False):
+    """
+    print("Extracting {} ...".format(lsb_file))
+
+    with open(lsb_file, "rb") as f:
+        data = f.read()
+    try:
+        lsb = LMScript.from_lsb(data)
+    except BadLsbError as e:
+        sys.exit("Failed to parse file: {}".format(e))
+
+    csv_data = []
+    scan_texts(lsb, csv_data, lsb_file, extract=True)
+
+    if len(csv_data) == 0:
+        sys.exit("No text data found.")
+
+    if Path(csv_file).exists():
+        if not overwrite and not append:
+            sys.exit("File {} already exists. Please use --overwrite or --append option.".format(csv_file))
+    elif append:
+        print("File {} does not exist, but --append specified. A new file will be created.".format(csv_file))
+        append = False
+
+    with open(csv_file, ("a" if append else "w"), newline="\n") as csvfile:
+        csv_writer = csv.writer(csvfile, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL)
+        if not append:
+            csv_writer.writerow(["Filename", "Command line", "Command subindex", "Original entry", "Translated entry"])
+        for row in csv_data:
+            csv_writer.writerow(row)
+
+    print("{} Text lines extracted.".format(len(csv_data)))
+
+
+@lmlsb.command()
+@click.argument("lsb_file", required=True, type=click.Path(exists=True, dir_okay="False"))
+@click.argument("csv_file", required=True, type=click.Path(exists=False, dir_okay="False"))
+@click.option("--no-backup", is_flag=True, default=False, help="Do not generate backup of original lsb file.")
+def inserttext(lsb_file, csv_file, no_backup):
+    """Apply translated texts from the given csv file on a given lsb file.
+    The csv file have to be created by the extracttext function and translations have to be added.
+    The original LSB file will be backed up to <lsb_file>.bak unless the
+    --no-backup option is specified.
+    """
+    print("Patching {} ...".format(lsb_file))
+
+    with open(lsb_file, "rb") as f:
+        data = f.read()
+    try:
+        lsb = LMScript.from_lsb(data)
+    except BadLsbError as e:
+        sys.exit("Failed to parse file: {}".format(e))
+
+    csv_data = []
+
+    with open(csv_file, newline="\n") as csvfile:
+        csv_reader = csv.reader(csvfile, delimiter=",", quotechar='"')
+        for row in csv_reader:
+            csv_data.append(row)
+
+    patched = scan_texts(lsb, csv_data, lsb_file, patch=True)
+
+    if not patched:
+        sys.exit("No text lines patched, so not need to change LSB file.")
+
+    if not no_backup:
+        print("Backing up original LSB.")
+        shutil.copyfile(str(lsb_file), "{}.bak".format(str(lsb_file)))
+    try:
+        new_lsb_data = lsb.to_lsb()
+        with open(lsb_file, "wb") as f:
+            f.write(new_lsb_data)
+        print("Wrote new LSB.")
+    except LiveMakerException as e:
+        sys.exit('Could not generate new LSB file: {}'.format(e))
+
+
+def scan_texts(lsb, csv_data, file_name, extract=False, patch=False):
```

Comment:
This function is way too big and should be broken up into smaller chunks. Stuff can go into it's own function even if that function is only called once. Basically you should not have python functions where you need 10+ levels of indentation, because it becomes really hard to read even after just a few levels of indentation.

the main reason the reformatter makes this code look particularly bad is because of the excessive levels of indentation

Reply:
Way too big? Really? This function perfectly fits my old small 1280x1024 screen.
Split this function into more small functions just to avoid intention levels would make the readability way more worse.
- you have to permanently jump with your eyes between the main function and the sub functions to understand what the whole construction is doing.
- you have to transport all variables between the functions, who are very many in this case!
- i do not know many about the internal python operations, but i guess placing subroutine calls into the inner loops will have a big negative performance impact.
I will not split this function. Take it as it is, rewrite it by yourself or drop the whole pull request.

Reply:
Fair enough. I certainly do appreciate the PR, and don't mind cleaning it up myself.

but mainly my issue was just that python projects generally have a standard coding style (pep8), and nested indentations this deep are for the most part considered bad practice.

Reply:
I agree, too many intentions are bad style. But things are need to be done. Adding stuff just to avoid intentions levels will make the whole thing more complicated. Keep things as simple as possible.